### ISSUE TITLE
Windows build broken - missing sigs, bad pointer diff

### ISSUE THREAD

**Issue by avih** [gwsw/less#430]
Apparently gcc 13 doesn't like ptrdiff between `char *` and `unsigned char *` and errors out at `WIN32getch` (regression of a6e9f31).

Also, in `screen.c` and `ttyin.c` it's complaining that `sigs` is undeclared (via `ABORT_SIGS`) (regression of a4816259).

The following diff fixes both issues for me:
```diff
diff --git a/screen.c b/screen.c
index 576d5fa..0928058 100644
--- a/screen.c
+++ b/screen.c
@@ -281,6 +281,7 @@ extern int no_vbell;
 extern int no_back_scroll;
 extern int no_init;
 extern int no_keypad;
+extern int sigs;
 extern int top_scroll;
 extern int quit_if_one_screen;
 extern int oldbot;
@@ -3016,7 +3017,7 @@ public char WIN32getch(void)
 		{
 			char *up = utf8;
 			put_wchar(&up, currentKey.unicode);
-			utf8_size = up - utf8;
+			utf8_size = up - (char *)utf8;
 			if (utf8_size == 0)
 				return '\0';
 			ascii = utf8[0];
diff --git a/ttyin.c b/ttyin.c
index 7844714..711a449 100644
--- a/ttyin.c
+++ b/ttyin.c
@@ -31,6 +31,7 @@ public int tty;
 #if LESSTEST
 public char *ttyin_name = NULL;
 #endif /*LESSTEST*/
+extern int sigs;
 
 #if !MSDOS_COMPILER
 static int open_tty_device(constant char* dev)
```

**Comment by avih** [comment]
For what it's worth, there have been several windows build brokenness over the years, some of them for a long time till it got fixed (e.g. v617 was broken with `init_poll not found`, which got fixed only in v630).

The [windows build](https://github.com/jftuga/less-Windows) repo only builds releases, so I imagine that windows build brokenness can go unnoticed for a while.

It might be useful to to add github CI builds (of each commit/PR) to at least ensure that the build is not broken on (some) supported platforms..

**Comment by gwsw** [comment]
Fixed in 81f8b0d15cb4aa6257644e2e29899e80c7946651.

I didn't know about the CI feature in github. That looks useful; I'll investigate it.

**Comment by avih** [comment]
Thanks. Confirmed fixed.
